### PR TITLE
React signIn

### PR DESCRIPTION
**Pull request by Abdisalan**
Added SignIn Page where user can register and choose LongNames

This is not hooked with the user config yet

### CODE REVIEW

**Comment by Abdisalan** [comment]
fixes #9

File: app/client/components/LongName.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import { prepareSignIn, addLongName } from '../../client/ts/signIn-utils';
+
+interface LongNameRegister {
+    longName: string
```
Comment (nicholasyan) [inline app/client/components/LongName.tsx]: Is this used anywhere in the component?

File: app/client/signup.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<html>
```
Comment (nicholasyan) [inline app/client/signup.html]: Is this just a testing page? Or is it used in the app?

File: app/client/ts/signIn-utils.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+export async function addLongName(): Promise<void> {
+    const inputField : HTMLInputElement = <HTMLInputElement> this.refs.longName;
+    const name: string = inputField.value;
+    if (name) {
+        await safeClient.dns.register(name).then(() => {
+            const errorMsg : HTMLElement = <HTMLElement> this.refs.error_msg;
+            errorMsg.innerHTML = `LongName ${name} successfully registered!`;
+        }, (err) => {
+            const errorMsg : HTMLElement = <HTMLElement> this.refs.error_msg;
+            errorMsg.innerHTML = "There was an error processing your request. Please choose a different LongName.";
```
Comment (nicholasyan) [inline app/client/ts/signIn-utils.ts:27]: Maybe in the future we can report some reason back to the user about why their LongName was rejected? e.g. it was too long or didn't meet <x> rule

(Doesn't need to be this iteration.)

Comment (Abdisalan) [reply app/client/ts/signIn-utils.ts:27]: currently, the launcher can only tell if you put in an existing longname or just explode saying there were multi errors processing your longname. But I think we can differentiate it

**Pull request merged**